Set deprecation warnings as errors (#7235)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/warnings.md

intro_text: "Use the --warn-error flag to promote all warnings to errors or --warn-error-options for granular control through options."
---

> Enabling `WARN_ERROR` config or setting the `--warn-error` flag will convert dbt warnings into errors. Any time dbt would normally warn, it will instead raise an error. Examples include `--select` criteria that selects no resources, deprecations, configurations with no associated models, invalid test configurations, or tests and freshness checks that are configured to return warnings.

Author: ```suggestion
Enabling `WARN_ERROR` config or setting the `--warn-error` flag will convert _all_ dbt warnings into errors. Any time dbt would normally warn, it will instead raise an error. Examples include `--select` criteria that selects no resources, deprecations, configurations with no associated models, invalid test configurations, or tests and freshness checks that are configured to return warnings.
```

- When `error` is set to `"all"` or `"*"`, the optional `warn` parameter can be set to exclude specific warnings from being treated as exceptions.
- Use the `silence` parameter to ignore warnings. To silence certain warnings you want to ignore, you can specify them in the `silence` parameter. This is useful in large projects where certain warnings aren't critical and can be ignored to keep the noise low and logs clean.

Here's how you can use the [`--warn-error-options`](#use---warn-error-options-for-targeted-warnings) flag to promote _specific_ warnings to errors:
- [Test warnings](/reference/resource-configs/severity) with the `--warn-error-options '{"error": ["LogTestResults"]}'` flag.
- Jinja [exception warnings](/reference/dbt-jinja-functions/exceptions#warn) with `--warn-error-options '{"error": ["JinjaLogWarning"]}'`.
- No nodes selected with `--warn-error-options '{"error": ["NoNodesForSelectionCriteria"]}'`.
- Adapter deprecation warnings with `--warn-error-options '{"error": ["AdapterDeprecationWarning"]}'`.

</VersionBlock>

<VersionBlock firstVersion="1.10">

> ### Set deprecation warnings as errors

Reviewer: Hmm, I think this is maybe overkill, and we could instead just have this as one bullet in "Promote all warnings to errors or target specific warnings" section

You can use the [`--warn-error-options`](#use---warn-error-options-for-targeted-warnings) flag to promote _deprecation_ warnings to errors, for example:

> - Set only deprecation warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`

Reviewer: same comment re: capitalization of `Deprecations`

Reviewer 2: ➕

> - Select only deprecations warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`

Reviewer 3: Should we say "set" instead of "select? Select doesn't feel right? We're actually making this happen right?
```suggestion
- Set only deprecation warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`
```

Author: If taken in context with the rest of the page (specifically the `--warn-err-options` section, they're not configuring a setting, but selecting which warnings to set as errors

> - Select only deprecations warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`

Author: ```suggestion
- Select only deprecation warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`
```

> - Select all deprecations except one and a specific warning as errors with `--warn-error-options="{'error': ['deprecations', 'SomeOtherWarning], 'exclude': ['DeprecationIDontWantErroring]}"`

Reviewer 3: ```suggestion
- Set all deprecations and a specific warning as errors and let one deprecation remain a warning with `--warn-error-options="{'error': ['deprecations', 'SomeOtherWarning], 'exclude': ['DeprecationIDontWantErroring]}"`
```

> - Select all warnings except deprecations as errors with `--warn-error-options="{'error': 'all', 'warn': ['deprecations']}"`

Reviewer 3: ```suggestion
- Set all warnings as errors except deprecations with `--warn-error-options="{'error': 'all', 'warn': ['deprecations']}"`
```

> - Select all warnings except deprecations as errors with `--warn-error-options="{'error': 'all', 'warn': ['deprecations']}"`

Author: ```suggestion
- Select all warnings as errors except deprecation warnings with `--warn-error-options="{'error': 'all', 'warn': ['deprecations']}"`
```

> - Select all warnings as errors except deprecation warnings with `--warn-error-options="{'error': 'all', 'warn': ['deprecations']}"`

Author: ```suggestion
- Set all warnings as errors except deprecation warnings with `--warn-error-options="{'error': 'all', 'warn': ['deprecations']}"`
```

> - Select a specific deprecation warning as an error while silencing all others with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`

Reviewer 3: ```suggestion
- Set a specific deprecation as an error while silencing all other deprecations with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`
```

Author: See the above feedback. They're also electing to change warnings to errors with this so I think we should leave some language in to be clear
```suggestion
- Select a specific deprecation warning as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`
```

> - Set a specific deprecation warning as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`

Author: ```suggestion
- Select a specific deprecation warning to set as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`
```

> - Select a specific deprecation warning as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`

Author: ```suggestion
- Set a specific deprecation warning as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`
```
